src/main/java/app/attestation/server/AttestationServer.java

```diff
@@ -1105,22 +1110,13 @@ private static void writeDevicesJson(final HttpExchange exchange, final long use
                 }
                 device.add("verifiedTimeFirst", select.columnLong(22));
                 device.add("verifiedTimeLast", select.columnLong(23));
-
-                final SQLiteStatement history = conn.prepare("SELECT time, strong, teeEnforced, " +
-                        "osEnforced FROM Attestations WHERE fingerprint = ? ORDER BY id DESC");
-                history.bind(1, select.columnBlob(0));
-
-                final JsonArrayBuilder attestations = Json.createArrayBuilder();
-                while (history.step()) {
-                    attestations.add(Json.createObjectBuilder()
-                            .add("time", history.columnLong(0))
-                            .add("strong", history.columnInt(1) != 0)
-                            .add("teeEnforced", history.columnString(2))
-                            .add("osEnforced", history.columnString(3)));
+                final SQLiteStatement devicesAttestationsLatestSelect = conn.prepare(
+                        "SELECT id FROM Attestations WHERE fingerprint = ? ORDER BY id DESC LIMIT 1");
```

Comment:
`SELECT MAX(id) FROM Attestations WHERE fingerprint = ?` would be more idiomatic and likely more efficient.